### PR TITLE
add ip_family config in harbor.yml

### PR DESCRIPTION
**Pull request by MinerYang**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
 https://github.com/goharbor/harbor/issues/19932

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by stonezdj — APPROVED** [review]

**Comment by zyyw** [comment]
@MinerYang this PR will be merge into main, and the next release against main will be `2.11.0`.
Please remove the label of `target/2.9.3` and add the label of `target/2.11.0`.
Thanks

File: make/harbor.yml.tmpl [hunk 1]
```diff
@@ -17,6 +17,16 @@ https:
   certificate: /your/certificate/path
   private_key: /your/private/key/path
 
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+# ip_family:
+#   # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
```
Comment (zyyw) [inline make/harbor.yml.tmpl:23]: change `ipv6Enabled` to `ipv6.enabled`?

File: make/harbor.yml.tmpl [hunk 2]
```diff
@@ -17,6 +17,16 @@ https:
...
+#   ipv6:
+#     enabled: false
+#   # ipv4Enabled set to true by default, currently it affected the nginx related component
```
Comment (zyyw) [inline make/harbor.yml.tmpl:26]: change `ipv4Enabled` to `ipv4.enabled`?

File: make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja [hunk 3]
```diff
@@ -33,6 +33,28 @@ https:
 #   private_key: /your/private/key/path
 {% endif %}
 
+{% if ip_family is defined %}
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+ip_family:
+  # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja:40]: same as above

File: make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja [hunk 4]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv6:
+    enabled: {{ ip_family.ipv6.enabled | lower }}
+  # ipv4Enabled set to true by default, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja:43]: change ipv4Enabled to ipv4.enabled?

File: make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja [hunk 5]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv4:
+    enabled: {{ ip_family.ipv4.enabled | lower }}
+{% else %}
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+ip_family:
+  # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja:50]: change ipv6Enabled to ipv6.enabled?

File: make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja [hunk 6]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv6:
+    enabled: false
+  # ipv4Enabled set to true by default, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_10_0/harbor.yml.jinja:53]: change ipv4Enabled to ipv4.enabled?

File: make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja [hunk 7]
```diff
@@ -33,6 +33,28 @@ https:
 #   private_key: /your/private/key/path
 {% endif %}
 
+{% if ip_family is defined %}
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+ip_family:
+  # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja:40]: change ipv6Enabled to ipv6.enabled?

File: make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja [hunk 8]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv6:
+    enabled: {{ ip_family.ipv6.enabled | lower }}
+  # ipv4Enabled set to true by default, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja:43]: change ipv4Enabled to ipv4.enabled?

File: make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja [hunk 9]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv4:
+    enabled: {{ ip_family.ipv4.enabled | lower }}
+{% else %}
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+ip_family:
+  # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
+  ipv6:
+    enabled: false
+  # ipv4Enabled set to true by default, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja:53]: change ipv4Enabled to ipv4.enabled?

File: make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja [hunk 10]
```diff
@@ -33,6 +33,28 @@ https:
...
+  ipv6:
+    enabled: {{ ip_family.ipv6.enabled | lower }}
+  # ipv4Enabled set to true by default, currently it affected the nginx related component
+  ipv4:
+    enabled: {{ ip_family.ipv4.enabled | lower }}
+{% else %}
+# # Harbor will set ipv4 enabled only by defualt if this block is not configured
+# # Otherwise, please uncomment this block to configure your own ip_family stacks
+ip_family:
+  # ipv6Enabled set to true if ipv6 is enabled in docker network, currently it affected the nginx related component
```
Comment (zyyw) [inline make/photon/prepare/migrations/version_2_9_0/harbor.yml.jinja:50]: change ipv6Enabled to ipv6.enabled?

**Comment by zyyw** [comment]
@MinerYang there is no `values.yaml` (the title of this PR: `add ip_family config in values.yaml`) for harbor offline-installer. Maybe change `values.yaml` to `harbor.yml`

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**